Marlin/src/pins/lpc1769/pins_FLY_CDY.h

```diff
@@ -0,0 +1,199 @@
+#ifndef E2_CS_PIN
+  #define E2_CS_PIN                        P0_10
+#endif
+
+//
+// Software SPI pins for TMC2130 stepper drivers
+//
+#if ENABLED(TMC_USE_SW_SPI)
+  #ifndef TMC_SW_MOSI
+    #define TMC_SW_MOSI                    P0_20
+  #endif
+  #ifndef TMC_SW_MISO
+    #define TMC_SW_MISO                    P0_19
+  #endif
+  #ifndef TMC_SW_SCK
+    #define TMC_SW_SCK                     P0_21
+  #endif
+#endif
+
+
+#if HAS_TMC_UART
+  #define X_SERIAL_TX_PIN                   P1_04
+  #define X_SERIAL_RX_PIN                   P1_04
+
+  #define Y_SERIAL_TX_PIN                   P1_10
+  #define Y_SERIAL_RX_PIN                   P1_10
+
+  #define Z_SERIAL_TX_PIN                   P1_16
+  #define Z_SERIAL_RX_PIN                   P1_16
+
+  #define E0_SERIAL_TX_PIN                  P4_28
+  #define E0_SERIAL_RX_PIN                  P4_28
+
+  #define E1_SERIAL_TX_PIN                  P2_12
+  #define E1_SERIAL_RX_PIN                  P2_12
+
+  #define E2_SERIAL_TX_PIN                  P0_10
+  #define E2_SERIAL_RX_PIN                  P0_10
+
+
+
+#endif
+
+
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN                      P0_26_A3  // (T4)
+#define TEMP_1_PIN                      P0_25_A2  // (T3)
+#define TEMP_2_PIN                      P0_24_A1  // (T2)
+#define TEMP_BED_PIN                    P0_23_A0  // (T1)
+
+//
+// Heaters / Fans
+//
+#define HEATER_BED_PIN                     P3_26
+#define HEATER_0_PIN                       P3_25
+#define HEATER_1_PIN                       P1_20
+#define HEATER_2_PIN                       P1_23
+#ifndef FAN_PIN
+  #define FAN_PIN                          P1_18
+#endif
+#define FAN1_PIN                           P1_21
+#define FAN2_PIN                           P1_24
+
+
+//
+// LCD / Controller
+//
+#define BEEPER_PIN                          P2_07
+#define LCD_PINS_RS                         P2_10
+#define LCD_PINS_ENABLE                     P0_22
+#define LCD_PINS_D4                         P1_19
+#define LCD_PINS_D5                         P2_08
+#define LCD_PINS_D6                         P1_30
+#define LCD_PINS_D7                         P1_31
+#define BTN_EN1                             P0_00
+#define BTN_EN2                             P0_01
+#define BTN_ENC                             P0_28
+
+
+#ifndef SDCARD_CONNECTION
+  #define SDCARD_CONNECTION              ONBOARD
+#endif
+    #define SS_PIN                          P0_06
+    #define SCK_PIN                         P0_07
+    #define MISO_PIN                        P0_08
+    #define MOSI_PIN                        P0_09
+    #define SD_DETECT_PIN                   P0_05
+#if SD_CONNECTION_IS(ONBOARD)
+
+  #endif
+
+ #if SD_CONNECTION_IS(LCD)
+
+  #define SCK_PIN                           P0_15
+  #define MISO_PIN                          P0_17
+  #define MOSI_PIN                          P0_18
+  #define SS_PIN                            P0_16
+  #define SD_DETECT_PIN                     P2_06
+
+#endif
```

Comment:
There are some issues with this block of defines. Is this what you meant?

```suggestion
#if SD_CONNECTION_IS(ONBOARD)
  #define SS_PIN                          P0_06
  #define SCK_PIN                         P0_07
  #define MISO_PIN                        P0_08
  #define MOSI_PIN                        P0_09
  #define SD_DETECT_PIN                   P0_05
#elif SD_CONNECTION_IS(LCD)
  #define SCK_PIN                         P0_15
  #define MISO_PIN                        P0_17
  #define MOSI_PIN                        P0_18
  #define SS_PIN                          P0_16
  #define SD_DETECT_PIN                   P2_06
#endif
```

Reply:
Yes. . I seem to make a mistake